src/ui/QueryForCommerce/QueryForCommerce.ts

```diff
@@ -0,0 +1,71 @@
+import { Component } from '../Base/Component';
+import { IComponentBindings } from '../Base/ComponentBindings';
+import { ComponentOptions } from '../Base/ComponentOptions';
+import { Assert } from '../../misc/Assert';
+import { QueryEvents, IBuildingQueryEventArgs } from '../../events/QueryEvents';
+import { AnalyticsEvents, IChangeAnalyticsCustomDataEventArgs } from '../../events/AnalyticsEvents';
+import { Initialization } from '../Base/Initialization';
+import { exportGlobally } from '../../GlobalExports';
+
+export interface IQueryForCommerceOptions {
+  listing?: string;
+}
+
+/**
+ * The QueryForCommerce component enables to set properties for listing pages in commerce sites in a convenient way
+ */
+export class QueryForCommerce extends Component {
+  static ID = 'QueryForCommerce';
```

Comment:
For the name...

Please @francoislg and @dlafreniere chime in. 

Not sure how it should be called. `QueryForCommerce` seems pretty awkward IMHO.

Think also that we might need to add other components with "Commerce" in their name, so, simply "CoveoCommerce" might not be a good idea...

Reply:
I think the `For` is a little awkward, so what about `CoveoCommerceQuery` 

It would be in line with `CoveoHiddenQuery` and kind of serves a similar purpose.

Reply:
CoveoCommerceQueryAddon ?

Reply:
> I think the `For` is a little awkward, so what about `CoveoCommerceQuery`
> 
> It would be in line with `CoveoHiddenQuery` and kind of serves a similar purpose.

+1

Reply:
Alrighty !